### PR TITLE
Fix for environment variable JELLYFIN_PublishedServerUrl being ignored.

### PR DESCRIPTION
**Pull request by BaronGreenback**
Fix for the second bug reported at https://github.com/jellyfin/jellyfin/issues/5280.

_GetSmartApiUrl_ and elsewhere reference __startupConfig.PublishedServerUrl_ however, this is only populated via the command-line and not via the environmental variables due to how the _IConfigurationBuilder_ is initialised.

The result is that JELLYFIN_PublishedServerUrl as specified at https://jellyfin.org/docs/general/administration/configuration.html doesn't work.

With this PR, PublishedServerUrl returns the correct value, with the command-line taking precedent.

### CODE REVIEW

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Now cmd args will get overwritten by everything else

**Comment by BaronGreenback** [comment]
@Bond-009 It shouldn't. 

The order of the addition into IConfigurationBuilder is still the same.

PublishedServerUrl is still selected if it contains a value, with the env var ignored.

The only difference _should_ be if the environmental variable JELLYFIN_PublishedServerUrl is defined. This will be saved in the startupConfig settings which is used by appHost.

**Comment by BaronGreenback** [comment]
> Now cmd args will get overwritten by everything else

Code has been re-designed and re-written.

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -231,6 +230,11 @@ public virtual bool CanLaunchWebBrowser
         /// </summary>
         public int HttpsPort { get; private set; }
 
+        /// <summary>
+        /// Gets the value of the PublishedServerUrl setting.
+        /// </summary>
+        public string PublishedServerUrl => _startupOptions.PublishedServerUrl ?? _startupConfig["PublishedServerUrl"];
```
Comment (Bond-009) [inline Emby.Server.Implementations/ApplicationHost.cs:236]: Why did this not work in the `ConvertToConfig`? And use the `UdpServer.AddressOverrideConfigKey` constant instead.

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/ApplicationHost.cs (hunk 1 above)
Comment (BaronGreenback) [reply Emby.Server.Implementations/ApplicationHost.cs:236]: Chicken and the egg situation. 'AddEnvironmentVariables' adds it to the config, but config is never passed through to 'ConvertToConfig'.

```
return config
                .SetBasePath(appPaths.ConfigurationDirectoryPath)
                .AddInMemoryCollection(inMemoryDefaultConfig)
                .AddJsonFile(LoggingConfigFileDefault, optional: false, reloadOnChange: true)
                .AddJsonFile(LoggingConfigFileSystem, optional: true, reloadOnChange: true)
                .AddEnvironmentVariables("JELLYFIN_")
                .AddInMemoryCollection(commandLineOpts.ConvertToConfig());
```

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**